### DISCUSSION TITLE
how to elevate privileges for Windows？

### DISCUSSION THREAD

**Discussion by pangzixiang** [tauri-apps/tauri#4201] [Q&A] [answered]
for example, I want to get the admin access when I start up the app

**Comment by joseph0x45** [comment]
I'm looking for that same functionality, didn't find anything so far

**Comment by FabianLars** [comment, marked as answer]
As said on Discord (if that was you?) normally the best practice here is to have a separate cli app, that does _only_ the things that need elevated permissions, which you'd invoke via the runas command.

Alternatively if you still want the whole app to be run with admin privileges, you can add a resource file into the exe, like described here for example <https://github.com/nabijaczleweli/rust-embed-resource#example-embedding-a-windows-manifest> (the winres crate should work too). Just make sure you run that after tauri-build (in buildrs), like this
```rs
fn main() {
    tauri_build::build();
    embed_resource::compile("your-appname.rc");
}
```

**Reply by joseph0x45** [reply]
I will try both of the solutions, thanks. And also, yes that was me on Discord

**Reply by xueweiwujxw** [reply]
```console
CVTRES : fatal error CVT1100: duplicate resource.  type:MANIFEST, name:1, language:0x0409
LINK : fatal error LNK1123: failure during conversion to COFF: file invalid or corrupt
```
I followed the steps in that example to build. But I found there is some error that Visual Studio may throw.
Here are my build files.
**`build.rs`**
```rust
extern crate embed_resource;
fn main() {
    tauri_build::build();
    embed_resource::compile("mkdat_bert_ui-manifest.rc");
}
```
**`mkdat_bert_ui-manifest.rc`**
```
#define RT_MANIFEST 24
1 RT_MANIFEST "mkdat_bert_ui.exe.manifest"
```
**`mkdat_bert_ui.exe.manifest`**
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
    <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
        <security>
            <requestedPrivileges>
                <requestedExecutionLevel level="requireAdministrator" uiAccess="false"/>
            </requestedPrivileges>
        </security>
    </trustInfo>
</assembly>
```

**Reply by FabianLars** [reply]
Yeah, `tauri-build` version 1.1.0 broke this, so you either need to downgrade to 1.0, wait for us to add proper manifest support (since it always was pure luck that it worked), or find a different solution to the elevation problem :/

**Reply by xueweiwujxw** [reply]
I got it. Thanks a lot.
I want administrator rights to write logs, so I changed the log directory to Temp to circumvent this problem

**Reply by larryteal** [reply]
@FabianLars tauri-build version 1.0  still error CVT1100 & LNK1123.

**Reply by FabianLars** [reply]
@larryteal did you lock it with the equal sign like `tauri-build = { version = "=1.0" }` ?

**Reply by larryteal** [reply]
@FabianLars Thanks for the reminder, it worked after adding the equal sign

**Reply by FabianLars** [reply]
tauri-build version 1.3 will officially support this: https://github.com/tauri-apps/tauri/pull/5730

**Reply by Danzo7** [reply]
> As said on Discord (if that was you?) normally the best practice here is to have a separate cli app, that does _only_ the things that need elevated permissions, which you'd invoke via the runas command.
I have a  separated CLI app that I run using`command` , How am I suppose to make this CLI app run with admin priveleges as it is spawned by tauri  which does not those priveleges I guess

**Comment by larryteal** [comment]
I want to set uiAccess="true", but program cannot start and I don't know where the problem is, does anyone share the successful experience of setting uiAccess="true"?

1. Created a test certificate and signed the program [（Discussions: Windows - Code signing）](https://github.com/tauri-apps/tauri/discussions/5739)
2. Install the program to a privileged directory (C:\Program Files)

**Program cannot start，why?**

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
    <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
        <security>
            <requestedPrivileges>
                <requestedExecutionLevel level="asInvoker" uiAccess="true"/>
            </requestedPrivileges>
        </security>
    </trustInfo>
</assembly>
```

**Reply by FabianLars** [reply]
Hmm, the only issue i could think of is that it doesn't like test certificates 🤔
